bson/_cbsonmodule.c

```diff
@@ -2797,6 +2807,86 @@ static PyObject* _cbson_decode_all(PyObject* self, PyObject* args) {
     return result;
 }
 
+
+static PyObject* _cbson_array_of_documents_to_buffer(PyObject* self, PyObject* args) {
+    uint32_t size;
+    uint32_t value_length;
+    uint32_t position = 0;
+    int32_t initial_position;
+    buffer_t buffer;
+    const char* string;
+    PyObject* arr;
+    PyObject* result = NULL;
+    Py_buffer view = {0};
+
+    if (!PyArg_ParseTuple(args, "O", &arr)) {
+        return NULL;
+    }
+
+    buffer = pymongo_buffer_new();
+
+
+    if (!_get_buffer(arr, &view)) {
+        goto fail;
+    }
+
+    string = (char*)view.buf;
+    memcpy(&size, string, 4);
```

Comment:
Missing a length check to guard against a zero length memoryview being passed in.

Reply:
isn't that what `total_size < BSON_MIN_SIZE` is checking?

Reply:
I don't see a `total_size < BSON_MIN_SIZE` check. We can't call memcpy without first validating that there's at least 4 bytes in the top-level buffer.

Reply:
It was down further, I moved it up